l1-contracts/contracts/bridgehub/ChainAssetHandler.sol

```diff
@@ -179,7 +275,17 @@ contract ChainAssetHandler is
             (BridgehubMintCTMAssetData)
         );
 
-        (address zkChain, address ctm) = BRIDGEHUB.forwardedBridgeMint(
+        uint256 currentMigrationNumber = migrationNumber[bridgehubMintData.chainId];
+        /// If we are not migrating for the first time, we check that the migration number is correct.
+        if (currentMigrationNumber != 0) {
+            require(
+                currentMigrationNumber + 1 == bridgehubMintData.migrationNumber,
```

Comment:
This line wouldnt work in the following sequence of events:
- Chain migrates to GW1
- Chain migrates to L1
- Chain migrates to GW2
- Chain migrates to L1
- Chain migrates to GW1 => will fail since `bridgehubMintData.migrationNumber` has been incremeneted too far.

I think we should do the following:
- If we are on L1, check the assert (as this would always work)
- If we are not on L1, trust the L1 value (as L1 is trusted to provide the correct number)

Reply:
https://github.com/matter-labs/era-contracts/pull/1491/commits/9984c3066278865cbbe38107f342dda49165cd75